oem/ibm/libpldmresponder/file_io_type_pcie.cpp

```diff
@@ -0,0 +1,148 @@
+#include "file_io_type_pcie.hpp"
+
+#include "libpldm/base.h"
+#include "oem/ibm/libpldm/file_io.h"
+
+#include "common/utils.hpp"
+#include "host-bmc/dbus/custom_dbus.hpp"
+
+#include <stdint.h>
+
+#include <iostream>
+
+namespace pldm
+{
+namespace responder
+{
+
+static constexpr auto pciePath = "/var/lib/pldm/pcie-topology/";
+constexpr auto topologyFile = "topology";
+constexpr auto cableInfoFile = "cableinfo";
+
+namespace fs = std::filesystem;
+std::map<uint16_t, bool> PCIeInfoHandler::receivedFiles;
+
+int PCIeInfoHandler::writeFromMemory(
+    uint32_t offset, uint32_t length, uint64_t address,
+    oem_platform::Handler* /*oemPlatformHandler*/)
+{
+    if (!fs::exists(pciePath))
+    {
+        fs::create_directories(pciePath);
+        fs::permissions(pciePath,
+                        fs::perms::others_read | fs::perms::owner_write);
+    }
+
+    fs::path infoFile(fs::path(pciePath) / topologyFile);
+    if (infoType == PLDM_FILE_TYPE_CABLE_INFO)
+    {
+        infoFile = (fs::path(pciePath) / cableInfoFile);
+    }
+
+    std::ofstream pcieData(infoFile, std::ios::out | std::ios::binary);
+    if (!pcieData)
+    {
+        std::cerr << "PCIe Info file creation error " << std::endl;
+        return -1;
+    }
+
+    auto rc = transferFileData(infoFile, false, offset, length, address);
+    if (rc != PLDM_SUCCESS)
+    {
+        std::cerr << "transferFileData failed with rc= " << rc << " \n";
+        return rc;
+    }
+
+    return PLDM_SUCCESS;
+}
+
+int PCIeInfoHandler::write(const char* buffer, uint32_t, uint32_t& length,
+                           oem_platform::Handler* /*oemPlatformHandler*/)
+{
+    int rc = PLDM_SUCCESS;
```

Comment:
reduce the scope of this variable. - this actually seems like a redundant var

Reply:
Done